sui_core/src/authority_batch.rs

```diff
@@ -193,6 +196,17 @@ impl crate::authority::AuthorityState {
                     .batches
                     .insert(&new_batch.batch.next_sequence_number, &new_batch)?;
 
+                // If a checkpointing service is present, register the batch with it
+                // to insert the transactions into future checkpoint candidates
+                if let Some(checkpoint) = &self._checkpoints {
+                    if let Err(err) = checkpoint
```

Comment:
```suggestion
                if let Some(Err(err)) = &self._checkpoints {
```

Reply:
Hm, I think we are calling a bunch of things on `checkpoint` on the second line, so not sure how to fold into 1 line.

Reply:
```rust
                if let Some(checkpoint @ Err(err)) = &self._checkpoints {
```
(from memory).

Reply:
Ok, I will let this be for the moment. The fragment of code is now:
```
                // If a checkpointing service is present, register the batch with it
                // to insert the transactions into future checkpoint candidates
                if let Some(checkpoint) = &self._checkpoints {
                    if let Err(err) = checkpoint
                        .lock()
                        .handle_internal_batch(new_batch.batch.next_sequence_number, &current_batch)
                    {
                        error!("Checkpointing service error: {}", err);
                    }
                }
```
So it is handy to first check if we have a checkpoint engine, and then lock it and call stuff on it. (The addition of the lock changed since the last review I think).